addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/generators/DataGenerator.java

```diff
@@ -96,7 +96,10 @@ private static String extractExample(Parameter parameter) {
                 .map(stream -> stream.map(Example::getValue).filter(Objects::nonNull).findFirst())
                 .orElse(Optional.ofNullable(parameter.getExample()))
                 .map(Object::toString)
-                .orElse(null);
+                .orElse(
+                        Optional.ofNullable(parameter.getSchema().getExample())
```

Comment:
Can't this be moved to after line 97? (And avoid duplicate the map and the orElse.)

Reply:
I had tried
```java
private static String extractExample(Parameter parameter) {
    return Optional.ofNullable(parameter.getExamples())
            .map(Map::values)
            .map(Collection::stream)
            .map(stream -> stream.map(Example::getValue).filter(Objects::nonNull).findFirst())
            .orElse(Optional.ofNullable(parameter.getExample()))
            .orElse(Optional.ofNullable(parameter.getSchema().getExample()))
            .map(Object::toString)
            .orElse(null);
}
```
but my IDE warned me that the `map` is illegal because the second orElse doesn't return an Optional. So I went with the current approach.

Reply:
Was thinking on `or` but that's not available in Java 8, as is then.